src/routes/terms.ts

```diff
@@ -0,0 +1,217 @@
+
+      res.status(200).send(terms);
+    } catch (err) {
+      next(err);
+    }
+  }
+);
+
+// get current term
+router.get(
+  "/current",
+  authorize(["ADMINISTRATOR", "PROFESSOR", "STUDENT"]),
+  async (req, res, next) => {
+    const currentDate = new Date()
+    try {
+      const term = await client.term.findFirstOrThrow({
+        where: { 
+          startTime: {
+            lt: currentDate
+          },
+          endTime : {
+            gt: currentDate
+          }
+        },
+      });
+
+      res.status(200).send(term);
+    } catch (err) {
+      next(err);
+    }
+  }
+);
+
+// get specified term
+router.get(
+  "/:termId",
+  authorize(["ADMINISTRATOR", "PROFESSOR", "STUDENT"]),
+  async (req, res, next) => {
+    try {
+      const term = await client.term.findUniqueOrThrow({
+        where: { id: req.params.termId },
+      });
+
+      res.status(200).send(term);
+    } catch (err) {
+      next(err);
+    }
+  }
+);
+
+// create new term
+router.post(
+  "/",
+  authorize(["ADMINISTRATOR"]),
+  async (req, res, next) => {
+    const schema = z.object({
+      season: z.nativeEnum(Season),
+      year: z.string().regex(new RegExp(/\d{4}/)),
+      startTime: z.string().datetime(),
+      endTime: z.string().datetime(),
+    })
+
+    try {
+      const body = schema.parse(req.body)
+      if (new Date(body.startTime) >= new Date(body.endTime)) {
+        res.sendStatus(400)
+        return
+      }
+      const conflicts = await client.term.findMany({
+        where: {
+          OR: [
+            {
+              // end time is within the new term's start and end time
+              endTime: {
+                gt: body.startTime,
+                lt: body.endTime
+              }
+            },
+            {
+              // start time is within the new term's start and end time
+              startTime: {
+                gt: body.startTime,
+                lt: body.endTime
+              }
+            },
+            {
+              // new term is entirely within
+              startTime: {
+                lt: body.startTime
+              },
+              endTime: {
+                gt: body.endTime
+              }
+            }
+          ]
+        }
+      })
+
+      if (conflicts.length == 0) {
+        const term = await client.term.create({data: body})
+        res.status(201).send(term)
+      } else {
+        res.sendStatus(400)
+        return
+      }
+    } catch(err) {
+      next(err)
+    }
+})
+
+// update term
+router.put("/:termId", authorize(["ADMINISTRATOR"]), async (req, res, next) => {
+  const schema = z.object({
+    season: z.nativeEnum(Season).optional(),
+    year: z.string().regex(new RegExp(/\d{4}/)).optional(),
+    startTime: z.string().optional(),
+    endTime: z.string().optional(),
+  })
+
+  try {
+    const body = schema.parse(req.body)
+    const term = await client.term.findUniqueOrThrow({
+      where: { id: req.params.termId }
+    })
+    const {id, ...data} = {
```

Comment:
The id can't be updated, so if you try passing it into the update function it'll cause a syntax error (even though it will always have the same value in this case). This just sets everything but the id into the data variable, which is passed into the update function.